src/jrd/tpc.cpp

```diff
@@ -364,7 +363,10 @@ void TipCache::mapInventoryPages(GlobalTpcHeader* header)
 	const TpcBlockNumber lastNumber = header->latest_transaction_id / m_transactionsPerBlock;
 
 	for (; blockNumber <= lastNumber; blockNumber++)
-		getTransactionStatusBlock(header, blockNumber);
+	{
+		Sync sync(&m_sync_status, FB_FUNCTION);
+		getTransactionStatusBlock(header, blockNumber, sync);
+	}
```

Comment:
Wouldn't it be better to move `Sync` outside of the loop?

Reply:
It should be unlocked before call of `getTransactionStatusBlock()` anyway. 
`mapInventoryPages()` run once, no need to optimize it in this place, imho